src/DIRAC/WorkloadManagementSystem/Utilities/RemoteRunner.py

```diff
@@ -219,5 +233,35 @@ def _wrapCommand(self, command, workingDirectory, executable):
             argumentsProcessed.append(os.path.join(".", os.path.basename(argument)))
 
         command = shlex.join(argumentsProcessed)
-        with open(executable, "w") as f:
+        with open(self.executable, "w") as f:
             f.write(command)
+            # Post-processing: compute the checksum of the outputs
+            f.write(f"\nmd5sum * > {self.checkSumOutput}")
+
+    def _checkOutputIntegrity(self, workingDirectory):
+        """Make sure that output files are not corrupted.
+
+        :param str workingDirectory: path of the outputs
+        """
+        checkSumOutput = os.path.join(workingDirectory, self.checkSumOutput)
+        if not os.path.exists(checkSumOutput):
+            return S_ERROR(f"Cannot guarantee the integrity of the outputs: {checkSumOutput} unavailable")
+
+        with open(checkSumOutput) as f:
+            # for each output file, compute the md5 checksum
+            for line in f:
+                checkSum, remoteOutput = list(filter(None, line.strip("\n").split(" ")))
+
+                hash = hashlib.md5()
+                localOutput = os.path.join(workingDirectory, remoteOutput)
+                if not os.path.exists(localOutput):
+                    return S_ERROR(f"{localOutput} was expected but not found")
+
+                with open(localOutput, "rb") as f:
+                    while chunk := f.read(128 * hash.block_size):
```

Comment:
Shouldn't this be 
```suggestion
                    while (chunk := f.read(128 * hash.block_size)):
```
?

Reply:
IIUC both ways are similar in this case. Would you prefer it with parenthesis?